src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -0,0 +1,375 @@
+from functools import partial
+from typing import Iterable, Union
+import numpy as np
+
+from braket.aws import AwsDevice
+from braket.devices import Device, LocalSimulator
+from braket.ahs.atom_arrangement import AtomArrangement
+from braket.ahs.analog_hamiltonian_simulation import AnalogHamiltonianSimulation
+from braket.ahs.driving_field import DrivingField
+from braket.timings.time_series import TimeSeries
+
+from pennylane import QubitDevice
+from pennylane._version import __version__
+from pennylane.pulse.rydberg_hamiltonian import RydbergHamiltonian, RydbergPulse
+
+
+class BraketAhsDevice(QubitDevice):
+    """Abstract Amazon Braket device for analogue hamiltonian simulation with PennyLane.
+
+    Args:
+        wires (int or Iterable[Number, str]]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers
+            (i.e., ``[-1, 0, 2]``) or strings (``['ancilla', 'q1', 'q2']``).
+        device (Device): The Amazon Braket device to use with PennyLane.
+        shots (int): Number of executions to run to aquire measurements. Defaults to 100.
+    """
+
+    name = "Braket AHS PennyLane plugin"
+    pennylane_requires = ">=0.29.0"
+    version = __version__
+    author = "Xanadu Inc."
+
+    operations = {"ParametrizedEvolution"}
+
+    def __init__(
+            self,
+            wires: Union[int, Iterable],
+            device: Device,
+            *,
+            shots=100):
+
+        if not shots:
+            raise RuntimeError(f"This device requires shots. Recieved shots={shots}")
+        self._device = backend
+
+        super().__init__(wires=wires, shots=shots)
+
+        self.register = None
+        self.ahs_program = None
+        self.samples = None
+
+    @property
+    def settings(self):
+        return {'interaction_coeff': 862690}  # MHz x um^6
+
+    def apply(self, operations, **kwargs):
+        """Convert the pulse operation to an AHS program and run on the connected device"""
+
+        if not np.all([op.name in self.operations for op in operations]):
+            raise NotImplementedError("Device {self.short_name} expected only operations "
+                                      "{self.operations} but recieved {operations}")
+
+        self._validate_operations(operations)
+
+        ev_op = operations[0]  # only one!
+
+        self._validate_pulses(ev_op.H.pulses)
+
+        ahs_program = self.create_ahs_program(ev_op)
+
+        task = self._run_task(ahs_program)
+
+        self.samples = task.result()
```

Comment:
Is this kind of shortcutting the execute pipeline, in the sense that there is no "measurement" and it is just setting the samples when the operation is applied and later accessing them with `get_samples`. Looks a bit hacky 🤔 

BTW I tried running it with `qml.expval(qml.PauliZ(0))` just to see what happens but I couldnt initialize the device (checked out this branch and pip-installed your plugin repo again)

![image](https://user-images.githubusercontent.com/43949391/227506340-933bbffd-dff9-45cf-b7be-f45c8bcd50ea.png)

Reply:
But it might still be a little buggy, I'm right in the middle of removing the jax dependencies and responding to a couple of other suggestions

Reply:
>Is this kind of shortcutting the execute pipeline, in the sense that there is no "measurement" and it is just setting the samples when the operation is applied and later accessing them with get_samples. Looks a bit hacky

This is a good point. I remember copying it from somewhere, but I don't know here, or whether/how that device implemented other types of measurement. I'll look into this as soon as the jax dependency is removed.